From 13c62892003786c22f160203040eb09040e96d66 Mon Sep 17 00:00:00 2001
From: Steve deRosier <Steve.Derosier@lairdtech.com>
Date: Sun, 12 Jul 2015 18:17:02 -0700
Subject: [PATCH] laird: Add custom gpio setup

Setup gpios to initial power-up values to avoid problems down the line.
Specifically:
* CHIP_PWD_L - set output and low to keep ath6003 in reset till we want it
* VBUS_SENSE - set input with no PU/PD so we don't resistor divider
* VBUS_EN - set output, low
* IRQ - set input, with a PU so we function correctly later

Bug: 7728, 7338, 7947
---
 board/at91sam9x5ek/at91sam9x5ek.c                 |   15 ++++++++++++++-
 board/at91sam9x5ek/at91sam9x5eknf_uboot_defconfig |    1 +
 2 files changed, 15 insertions(+), 1 deletion(-)

diff --git a/board/at91sam9x5ek/at91sam9x5ek.c b/board/at91sam9x5ek/at91sam9x5ek.c
index a026b92..745b672 100644
--- a/board/at91sam9x5ek/at91sam9x5ek.c
+++ b/board/at91sam9x5ek/at91sam9x5ek.c
@@ -46,7 +46,20 @@
 #include "board_hw_info.h"
 
 #ifdef CONFIG_USER_HW_INIT
-extern void hw_init_hook(void);
+static void hw_init_hook(void)
+{
+	/* Setup WB45NBT custom GPIOs */
+	const struct pio_desc gpios[] = {
+	/*  {"NAME",        PIN NUMBER,     VALUE, ATTRIBUTES, TYPE },*/
+		{"CHIP_PWD_L",	AT91C_PIN_PA(28),	0, PIO_DEFAULT, PIO_OUTPUT},
+		{"VBUS_SENSE",	AT91C_PIN_PB(11),	0, PIO_DEFAULT, PIO_INPUT},
+		{"VBUS_EN",	    AT91C_PIN_PB(12),	0, PIO_DEFAULT, PIO_OUTPUT},
+		{"IRQ",	        AT91C_PIN_PB(18),	0, PIO_PULLUP,  PIO_INPUT},
+		{(char *)0,	0, 0, PIO_DEFAULT, PIO_PERIPH_A},
+	};
+
+	pio_configure(gpios);
+}
 #endif
 
 static void at91_dbgu_hw_init(void)
diff --git a/board/at91sam9x5ek/at91sam9x5eknf_uboot_defconfig b/board/at91sam9x5ek/at91sam9x5eknf_uboot_defconfig
index 94e6220..8da3c1f 100644
--- a/board/at91sam9x5ek/at91sam9x5eknf_uboot_defconfig
+++ b/board/at91sam9x5ek/at91sam9x5eknf_uboot_defconfig
@@ -12,3 +12,4 @@ CONFIG_DEBUG_INFO=y
 CONFIG_THUMB=y
 CONFIG_SCLK=y
 CONFIG_LOAD_ONE_WIRE=n
+CONFIG_USER_HW_INIT=y
-- 
1.7.9.5

